auth/lib/utils/directory_utils.dart

```
@@ -1,12 +1,106 @@
+  static migrateNamingChanges() async {
+    try {
+      final sharedPrefs = await SharedPreferences.getInstance();
+      if (sharedPrefs.containsKey(migratedNamingChanges)) {
+        return;
+      }
+      final databaseFile = File(
+        p.join(
           (await getApplicationDocumentsDirectory()).path,
           "ente",
-          ".$databaseName",
-        ],
+          ".ente.authenticator.db",
+        ),
       );
+      final offlineDatabaseFile = File(
+        p.join(
+          (await getApplicationDocumentsDirectory()).path,
+          "ente",
+          ".ente.offline_authenticator.db",
+        ),
+      );
+      Directory oldDataDir;
+      Directory newDataDir;
+
+      if (Platform.isLinux) {
+        oldDataDir = Directory(
+          p.join(dataHome.path, "ente_auth"),
+        );
+      } else {
+        oldDataDir = Directory(
+          p.join(
+            (await getApplicationDocumentsDirectory()).path,
+            "ente",
+          ),
+        );
+      }
+      newDataDir = await getApplicationSupportDirectory();
+      await newDataDir.create(recursive: true);
+
+      File newDatabaseFile =
+          File(p.join(newDataDir.path, ".ente.authenticator.db"));
+      if (await databaseFile.exists() && !await newDatabaseFile.exists()) {
+        await databaseFile.copy(newDatabaseFile.path);
+      }
+
+      File newOfflineDatabaseFile =
+          File(p.join(newDataDir.path, ".ente.offline_authenticator.db"));
+      if (await offlineDatabaseFile.exists() &&
+          !await newOfflineDatabaseFile.exists()) {
+        await offlineDatabaseFile.copy(newOfflineDatabaseFile.path);
+      }
+
+      if (Platform.isLinux && await oldDataDir.exists()) {
+        await copyPath(oldDataDir.path, newDataDir.path);
+      }
+
+      sharedPrefs.setBool(migratedNamingChanges, true).ignore();
+    } catch (e, st) {
+      logger.warning("Migrating Database failed!", e, st);
```

Should we rethrow the exception here?

---

Done!